src/codeclimate.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+use rustfmt_diff::{Mismatch, DiffLine};
+use std::io::{self, Write};
+
+pub fn output_codeclimate<T>(mut writer: T,
+                             filename: &str,
+                             diff: Vec<Mismatch>)
+                             -> Result<(), io::Error>
+    where T: Write
+{
+    for mismatch in diff {
+        for line in mismatch.lines {
+            if let DiffLine::Expected(ref exp) = line {
+                try!(write!(writer,
+                            "{{\"type\": \"issue\",\
+                    \"check_name\":  \"Style\",\
+                    \"description\": \"Should be `{exp}`\",\
+                    \"categories\":  [\"Style\"],\
+                    \"location\": {{\
+                        \"path\": \"{filename}\",\
+                        \"lines\": {{\
+                            \"begin\": {line},\
+                            \"end\":   {line}\
+                        }}\
+                    }}\
+                }}\r\n\0",
```

Comment:
@nrc   using serde_json will likely change the rust versions that rustfmt compiles for because of custom derive macros. are you sure you want me to introduce it?   i tries using rustc_serialize::json but that does not allow renaming fields, so the "type" field name cannot be produced as output.

Either way i'd have to make the change much more invasive to avoid string fmt here. Should i go ahead anyway?